spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural.py

```diff
@@ -0,0 +1,514 @@
+import numpy as np
+import collections
+
+from spinn_utilities.overrides import overrides
+from data_specification.enums.data_type import DataType
+from spynnaker.pyNN.models.neural_projections import ProjectionApplicationEdge
+from spynnaker.pyNN.models.neural_projections import ProjectionMachineEdge
+from .abstract_plastic_synapse_dynamics import AbstractPlasticSynapseDynamics
+from .abstract_synapse_dynamics_structural import \
+    AbstractSynapseDynamicsStructural
+from .synapse_dynamics_stdp import SynapseDynamicsSTDP
+from .synapse_dynamics_static import SynapseDynamicsStatic
+from spynnaker.pyNN.utilities import constants
+
+
+class SynapseDynamicsStructural(AbstractSynapseDynamicsStructural):
+    def __init__(self, stdp_model=None, f_rew=10 ** 4, weight=0, delay=1,
+                 s_max=32,
+                 sigma_form_forward=2.5, sigma_form_lateral=1,
+                 p_form_forward=0.16, p_form_lateral=1,
+                 p_elim_dep=0.0245, p_elim_pot=1.36 * 10 ** -4,
+                 grid=np.array([16, 16]), lateral_inhibition=0,
+                 random_partner=False,
+                 seed=None):
+
+        AbstractSynapseDynamicsStructural.__init__(self)
+        self._f_rew = f_rew  # Hz
+        self._p_rew = 1. / self._f_rew  # ms
+        self._weight = weight
+        self._delay = delay
+        self._s_max = s_max  # maximum number of presynaptic neurons
+        self._lateral_inhibition = lateral_inhibition
+        self._sigma_form_forward = sigma_form_forward
+        self._sigma_form_lateral = sigma_form_lateral
+        self._p_form_forward = p_form_forward
+        self._p_form_lateral = p_form_lateral
+        self._p_elim_dep = p_elim_dep
+        self._p_elim_pot = p_elim_pot
+        self._grid = np.asarray(grid, dtype=int)
+        self._random_partner = random_partner
+        self._connections = {}
+
+        self.fudge_factor = 1.5
+        self._actual_row_max_length = self._s_max
+
+        if stdp_model is not None:
+            self.super = SynapseDynamicsSTDP(
```

Comment:
Can we _PLEASE_ call this property something else that doesn't conflict with a Python builtin? I know it impacts on other files… but using the same name as a builtin is just asking for confused programmers! (Perhaps `underlying` would do for a name?)

Reply:
self.__super would be fine, or even calling it something else. I don't really care either way